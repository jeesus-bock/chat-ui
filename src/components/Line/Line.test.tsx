import { describe, expect, test } from 'vitest';
import { render } from 'solid-testing-library';
import { Line } from './index';

describe('<Line />', () => {
  test('renders', () => {
    const { container, unmount } = render(() => <Line msg={{ type: 'msg', from: 'mock_user', to: 'mock_room', msg: 'This is a test message!', ts: 165789872275 }} />);
    expect(container).toMatchSnapshot();
    unmount();
  });

  test('present message data', async () => {
    const { queryByTestId, unmount } = render(() => <Line msg={{ type: 'msg', from: 'mock_user', to: 'mock_room', msg: 'This is a test message!', ts: 165789872275 }} />);
    expect(queryByTestId('ts')).toHaveTextContent('10:44');
    expect(queryByTestId('from')).toHaveTextContent('mock_user');
    expect(queryByTestId('msg')).toHaveTextContent('This is a test message!');
  });
});
