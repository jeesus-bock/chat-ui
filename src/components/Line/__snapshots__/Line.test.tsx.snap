// Vitest Snapshot v1

exports[`<Line /> > renders 1`] = `
<div>
  <div
    class="flex px-4 _cols_cesd7_1 grid"
  >
    <span
      class="inline-block pr-4 mr-4 border-r"
      data-testid="ts"
    >
      10:44
    </span>
    <span
      class="ml-2 pr-4 mr-4 border-r ml-0"
      data-testid="from"
    >
      <!--#-->
      mock_user
      <!--/-->
      <!--#-->
      
      <!--/-->
    </span>
    <span
      class="ml-2 ml-0 border-r"
      data-testid="msg"
    >
      This is a test message!
    </span>
  </div>
</div>
`;
